src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.EmptyFieldException;
+import core.basesyntax.exception.IllegalAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.LoginExistingException;
```

Comment:
- You should create your custom unchecked exception and throw it in case of invalid data.

I think all this exceptions are redundant because all of them extends from `RuntimeException` and every one of them just throwing exception class name and message.

You could make one exception for invalid data in general.